django/core/handlers/asgi.py

```diff
@@ -239,61 +251,66 @@ def handle_uncaught_exception(self, request, resolver, exc_info):
                 content_type="text/plain",
             )
 
-    async def send_response(self, response, send):
+    async def send_response(self, response, send, receive):
         """Encode and send a response out over ASGI."""
         # Collect cookies into headers. Have to preserve header case as there
         # are some non-RFC compliant clients that require e.g. Content-Type.
-        response_headers = []
-        for header, value in response.items():
-            if isinstance(header, str):
-                header = header.encode("ascii")
-            if isinstance(value, str):
-                value = value.encode("latin1")
-            response_headers.append((bytes(header), bytes(value)))
-        for c in response.cookies.values():
-            response_headers.append(
-                (b"Set-Cookie", c.output(header="").encode("ascii").strip())
-            )
-        # Initial response message.
-        await send(
-            {
-                "type": "http.response.start",
-                "status": response.status_code,
-                "headers": response_headers,
-            }
-        )
-        # Streaming responses need to be pinned to their iterator.
-        if response.streaming:
-            # - Consume via `__aiter__` and not `streaming_content` directly, to
-            #   allow mapping of a sync iterator.
-            # - Use aclosing() when consuming aiter.
-            #   See https://github.com/python/cpython/commit/6e8dcda
-            async with aclosing(aiter(response)) as content:
-                async for part in content:
-                    for chunk, _ in self.chunk_bytes(part):
-                        await send(
-                            {
-                                "type": "http.response.body",
-                                "body": chunk,
-                                # Ignore "more" as there may be more parts; instead,
-                                # use an empty final closing message with False.
-                                "more_body": True,
-                            }
-                        )
-            # Final closing message.
-            await send({"type": "http.response.body"})
-        # Other responses just need chunking.
-        else:
-            # Yield chunks of response.
-            for chunk, last in self.chunk_bytes(response.content):
-                await send(
-                    {
-                        "type": "http.response.body",
-                        "body": chunk,
-                        "more_body": not last,
-                    }
+        try:
+            await asyncio.wait_for(self.listen_for_disconnect(receive), timeout=0.05)
```

Comment:
wait for 0.05 seconds and only continue if self.listen_for_disconnect times out.

Reply:
I don't think pausing here is the way forward. The happy path is no disconnect, and we're adding a quite big delay to check for the deviant case. 

We should be able to use [get_nowait](https://docs.python.org/3.11/library/asyncio-queue.html#asyncio.Queue.get_nowait) in this context no? 🤔

(But see overview comment: I'm not sure this is the right approach.)